packages/itwinui-css/src/progress-indicator/radial.scss

```diff
@@ -4,148 +4,109 @@
-    img {
-      @include mixins.iui-icon-style('m');
-      fill: var(--iui-color-icon);
-
-      &:hover {
-        fill: var(--iui-color-icon-hover);
-      }
-    }
+  :is(svg, img) {
+    @include mixins.iui-icon-style('m');
+    fill: var(--iui-color-icon);
   }
 
-  &.iui-positive {
-    color: var(--iui-color-text-positive);
+  @each $status in positive, negative {
+    &[data-iui-status='#{$status}'] {
+      color: var(--iui-color-text-#{$status});
 
-    svg.iui-radial {
-      circle.iui-fill,
-      circle.iui-track {
-        stroke: var(--iui-color-border-positive);
+      &::after {
+        background-color: var(--iui-color-border-#{$status});
       }
-    }
 
-    .iui-inner-content svg {
-      fill: var(--iui-color-icon-positive);
+      :is(svg, img) {
+        fill: var(--iui-color-border-#{$status});
+      }
     }
   }
 
-  &.iui-negative {
-    color: var(--iui-color-text-negative);
-
-    svg.iui-radial circle.iui-fill {
-      stroke: var(--iui-color-border-negative);
-    }
+  &[data-iui-size='x-small'] {
+    --_iui-progress-indicator-radial-size: #{map.get(mixins.$iui-icon-sizes, m)};
 
-    .iui-inner-content svg {
-      fill: var(--iui-color-icon-negative);
+    &::before,
+    &::after {
+      border-width: var(--iui-size-3xs);
     }
-  }
-
-  &.iui-x-small {
-    --_iui-progress-indicator-radial-size: #{map.get(mixins.$iui-icon-sizes, m)};
 
-    .iui-inner-content {
+    > * {
       display: none;
     }
   }
 
-  &.iui-small {
+  &[data-iui-size='small'] {
     --_iui-progress-indicator-radial-size: #{map.get(mixins.$iui-icon-sizes, l)};
+
+    &::before,
+    &::after {
+      border-width: calc(var(--iui-size-3xs) * 1.5);
+    }
   }
 
-  &.iui-large {
+  &[data-iui-size='large'] {
     --_iui-progress-indicator-radial-size: #{map.get(mixins.$iui-icon-sizes, 2xl)};
+    font-size: var(--iui-font-size-2);
 
-    .iui-inner-content {
-      font-size: var(--iui-font-size-2);
+    &::before,
+    &::after {
+      border-width: calc(var(--iui-size-2xs) * 1.5);
+    }
 
-      svg,
-      img {
-        @include mixins.iui-icon-style('l');
-      }
+    :is(svg, img) {
+      @include mixins.iui-icon-style('l');
     }
   }
 }
 
 @mixin iui-progress-indicator-radial-determinate {
-  svg.iui-radial {
-    transform: rotate(-90deg);
-
-    circle.iui-fill {
-      // C = 2πR
-      stroke-dasharray: 88;
-      transition: stroke-dashoffset 0.5s ease-in-out;
-    }
+  &::after {
+    $percentage: var(--iui-progress-percentage, 100%);
+    mask-image: conic-gradient(#000 $percentage, #0000 $percentage), linear-gradient(#000 0 0);
   }
 }
 
 @mixin iui-progress-indicator-radial-indeterminate {
-  svg.iui-radial {
-    animation: rotate-indeterminate 2s linear infinite;
-
-    circle.iui-fill {
-      animation: dash-indeterminate 1.5s ease-in-out infinite;
-    }
-  }
+  &::after {
+    mask-image: conic-gradient(#0000, #000), linear-gradient(#000 0 0);
+    animation: rotate-indeterminate 0.8s linear infinite;
```

Comment:
we might want to use a custom cubic-bezier easing instead of `linear`

Reply:
also, because this is a gradient now, i'm not exactly sure how to handle forced-colors mode 🤔

Reply:
I like this slight ease-in-out cubic-bezier: https://cubic-bezier.com/#.6,.4,.4,.6

Reply:
And yes, forced-colors mode is a bit strange. It doesn't work at all for linnear 
<img width="389" alt="image" src="https://github.com/iTwin/iTwinUI/assets/83585998/0ae9ad63-7499-4ad6-9639-642b8d73dd39">

Reply:
Changed the easing

https://github.com/iTwin/iTwinUI/assets/9084735/af7661e0-36d0-41f7-b128-66e5201157a3